src/overlays/actors/ovl_Airplane/ac_airplane.c

```diff
@@ -2,64 +2,334 @@
+    static s16 goal_angle_random[8] = {
+        0x093E, 0x07D2, 0xFBBC, 0xF6C2, 0x0000, 0x1110, 0xEFA6, 0x105A,
+    };
+    static s16 goal_angle_random2[8] = {
+        0x31C4, 0x1FFE, 0xDC74, 0xE002, 0xD558, 0x2AA8, 0x238C, 0x0000,
+    };
+    UNUSED UNK_TYPE1 pad[0x4];
+    s32 joystickY = getJoystick_Y();
+    Player* player = get_player_actor_withoutCheck(game_play);
+    s32 randIndex = RANDOM_F(8);
+    s16* goalAngle;
+    UNUSED UNK_TYPE1 pad2[0x4];
+    s32 joystickYMagnitude;
+
+    if ((joystickY - this->joystickY) > 30) {
+        joystickYMagnitude = ABS(this->joystickY);
+        this->speed = ABS_F((f32)this->joystickY) * (7.0f / 80.0f) + 3.0f;
+
+        aAp_SelectFly(this, joystickYMagnitude);
+        goalAngle = (joystickYMagnitude > 50) ? goal_angle_random2 : goal_angle_random;
+        this->actor.world.pos.y = player->actor.world.pos.y + 25.0f;
+        this->actor.world.pos.x = player->actor.world.pos.x;
+        this->actor.world.pos.z = player->actor.world.pos.z;
+        this->yRot = RAD_TO_DEG(BINANG_TO_RAD(player->actor.shape.rot.y));
+        this->targetYRot = RAD_TO_DEG(BINANG_TO_RAD(goalAngle[randIndex])) + this->yRot;
+        this->xRot = this->speed * -(60.0f / 7.0f) + 30.0f;
+        this->zRot = 0.0f;
+        this->ySpeed = 0.0f;
+        mPlib_request_main_wait_type3(game_play);
+    }
+}
+
+void aAp_PlayerCatch(UNUSED Airplane* this, UNUSED Game_Play* game_play) {
+}
+
+void aAp_StopFlyMove(Airplane* this, UNUSED Game_Play* game_play) {
+    this->speed = 0.0f;
+    if ((this->actor.xzDistToPlayer <= 30.0f) && chkTrigger(A_BUTTON)) {
+        this->state = Ap_STATE_PLAYER_CATCH;
+    }
+}
+
+void aAp_FallFlyMove(Airplane* this, Game_Play* game_play) {
+    aAp_RubberMove(&this->xRot, 60.0f, 0.04f);
+    aAp_RubberMove(&this->speed, 5.0f, 0.02f);
+    this->actor.world.pos.y -= 0.15f;
+    if (this->actor.colResult.onGround) {
+        common_data.clip.unk_090->unk_00(1, this->actor.world.pos, 1, this->actor.world.rot.y, game_play,
+                                         this->actor.fgName, 0, 8);
+        this->state = Ap_STATE_STOP_FLY_MOVE;
+    }
+}
+
+void aAp_FallFlyMove2(Airplane* this, Game_Play* game_play) {
+    aAp_RubberMove(&this->xRot, 70.0f, 0.15f);
+    aAp_RubberMove(&this->speed, 10.0f, 0.02f);
+    this->actor.world.pos.y -= 0.15f;
+    if (this->actor.colResult.onGround) {
+        common_data.clip.unk_090->unk_00(1, this->actor.world.pos, 1, this->actor.world.rot.y, game_play,
+                                         this->actor.fgName, 0, 8);
+        this->state = Ap_STATE_STOP_FLY_MOVE;
+    }
+}
+
+void aAp_LeanAirplane(Airplane* this) {
+    if (this->state == Ap_STATE_FREE_FLY_MOVE) {
+        aAp_RubberMove(&this->zRot, 0.0f, 0.05f);
+    } else if (this->yRot < this->startYRot) {
+        aAp_RubberMove(&this->zRot, 50.0f, 0.05f);
+    } else {
+        aAp_RubberMove(&this->zRot, -50.0f, 0.05f);
+    }
+}
+
+static xyz_t wind1 = { 2.0f, 1.0f, 4.0f };
```

Comment:
Hmm non in function data has no static qualifier, and this symbol name doesn't repeat in the maps either

Reply:
ah ok will change!